src/view/sidebar/edit-modal/parts/main-edit-modal.tsx

```diff
@@ -134,89 +137,95 @@ type MainEditModalProps = {
 	grant: Grant[];
 };
 
-type EditModalContexType = {
-	setModal: (modal: string) => void;
-	onClose: () => void;
-	setActiveGrant: (grant: Grant) => void;
-};
-
 export const MainEditModal: FC<MainEditModalProps> = ({ folder, totalAppointments, grant }) => {
 	const allCalendars = useFoldersArray();
 
-	const iconColor = setCalendarColor({ color: folder.color, rgb: folder.rgb });
 	const [t] = useTranslation();
+	const accounts = useUserAccounts();
+	const createSnackbar = useSnackbar();
+	const dispatch = useAppDispatch();
+	const { setModal, onClose, setActiveGrant } = useEditModalContext();
+
 	const colors = useMemo(() => getStatusItems(t), [t]);
 
-	const [inputValue, setInputValue] = useState(folder.name || '');
-	const [freeBusy, setFreeBusy] = useState(false);
-	const toggleFreeBusy = useCallback(() => setFreeBusy((c) => !c), []);
-	const dispatch = useAppDispatch();
-	const checked = useMemo(() => folder.checked, [folder]);
+	const defaultFreeBusy = /b/.test(folder.f ?? '');
+	const defaultFolderName = folder.name || '';
+	const defaultColor = useMemo(
+		() =>
+			find(colors, { label: setCalendarColor({ color: folder.color, rgb: folder.rgb }).label }) ?? {
+				label: '',
+				value: ''
+			},
+		[colors, folder?.color, folder?.rgb]
+	);
 
-	const { setModal, onClose, setActiveGrant } = useContext<EditModalContexType>(EditModalContext);
-	const accounts = useUserAccounts();
-	const createSnackbar = useContext(SnackbarManagerContext);
-	const [hovered, setHovered] = useState({});
+	const [folderName, setFolderName] = useState(defaultFolderName);
+	const [freeBusy, setFreeBusy] = useState(defaultFreeBusy);
 
-	const onMouseEnter = useCallback((item) => {
-		setHovered(item);
-	}, []);
-	const onMouseLeave = useCallback(() => {
-		setHovered({});
-	}, []);
+	const toggleFreeBusy = useCallback(() => setFreeBusy((c) => !c), []);
 
 	const folderArray = useMemo(
 		() =>
 			map(allCalendars, (f) =>
-				f.name === folder.name || (f.view !== FOLDER_VIEW.appointment && parseInt(f.id, 10) > 16)
+				f.name === defaultFolderName ||
+				(f.view !== FOLDER_VIEW.appointment && parseInt(f.id, 10) > 16)
```

Comment:
I know that this is existing code, but the parseInt condition is cryptic. IMHO this can be encapsulated in a local function with a self descriptive name